Sources/SKSwiftPMWorkspace/SwiftPMWorkspace.swift

```diff
@@ -366,7 +369,7 @@ extension SwiftPMWorkspace {
     args += ["-c"]
     args += td.sources.map { $0.pathString }
     args += ["-I", buildPath.pathString]
-    args += td.compileArguments()
+    args += try! td.compileArguments()
```

Comment:
Why do we know this cannot throw here?  Same question for `basicArguments` below.

Reply:
to be clear I am happy to bring these changes back in, but they end up effecting things like the BuildSystem protocol (registerForChangeNotifications needs to handle errors or marked as throwing) which is a bit beyond my understanding of LSP

Reply:
I think errors from here are rare, but fatalError seems wrong since it will cause a crash. logging is fine I guess. would you like me to do that?

Reply:
see https://github.com/apple/sourcekit-lsp/pull/433/commits/4c36e5a322e51fc9dceef334ba376b0ba1e7fc6a